fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStoreTest.java

```diff
@@ -276,4 +279,69 @@ public void withToProto() throws Exception {
+            openMetaDataStore(context);
+            assertNotNull(metaDataStore.getRecordMetaData().getRecordType("MySimpleRecord"));
+            metaDataStore.addIndex("MySimpleRecord", "testIndex", "rec_no");
+            assertNotNull(metaDataStore.getRecordMetaData().getIndex("testIndex"));
+            context.commit();
+            assertNotNull(metaDataStore.getRecordMetaData().getIndex("testIndex"));
+        }
+
+        try (FDBRecordContext context = fdb.openContext()) {
+            openMetaDataStore(context);
+            assertNotNull(metaDataStore.getRecordMetaData().getIndex("testIndex"));
+            assertNotNull(metaDataStore.getRecordMetaData().getRecordType("MySimpleRecord"));
+            MetaDataException t = assertThrows(MetaDataException.class, () ->
+                    metaDataStore.addIndex("MySimpleRecord", "testIndex", "rec_no"));
+            assertEquals("Index testIndex already defined", t.getMessage());
+            metaDataStore.dropIndex("testIndex");
+            context.commit();
+            t = assertThrows(MetaDataException.class, () ->
+                    metaDataStore.getRecordMetaData().getIndex("testIndex"));
+            assertEquals("Index testIndex not defined", t.getMessage());
+        }
+
+        try (FDBRecordContext context = fdb.openContext()) {
+            openMetaDataStore(context);
+            assertNotNull(metaDataStore.getRecordMetaData().getRecordType("MySimpleRecord"));
+            MetaDataException t = assertThrows(MetaDataException.class, () ->
+                    metaDataStore.getRecordMetaData().getIndex("testIndex"));
+            assertEquals("Index testIndex not defined", t.getMessage());
+            t = assertThrows(MetaDataException.class, () -> metaDataStore.dropIndex("testIndex"));
+            assertEquals("No index named testIndex defined", t.getMessage());
+            context.commit();
+        }
+    }
+
+    @Test
+    public void updateRecords() throws Exception {
+        try (FDBRecordContext context = fdb.openContext()) {
+            openMetaDataStore(context);
+            RecordMetaData metaData = RecordMetaData.build(TestRecords1Proto.getDescriptor());
+            metaDataStore.saveRecordMetaData(metaData);
+            context.commit();
+            assertNotNull(metaDataStore.getRecordMetaData().getRecordType("MySimpleRecord"));
+        }
+
+        try (FDBRecordContext context = fdb.openContext()) {
+            openMetaDataStore(context);
+            assertNotNull(metaDataStore.getRecordMetaData().getRecordType("MySimpleRecord"));
+            metaDataStore.updateRecords(TestRecords1EvolvedProto.getDescriptor());
+            assertNotNull(metaDataStore.getRecordMetaData().getRecordType("MySimpleRecord"));
+            assertNotNull(metaDataStore.getRecordMetaData().getRecordType("MyOtherRecord"));
+            assertNotNull(metaDataStore.getRecordMetaData().getRecordType("AnotherRecord"));
+            context.commit();
+        }
```

Comment:
Might be worth creating a new meta-data store in a new transaction so that it doesn't just read the meta-data from cache. (It looks like the `indexes` tests are already doing that.)

Reply:
good idea. sure.